src/endpoints/getEvent.ts

```diff
@@ -39,13 +40,41 @@ export const getEvent = (config: HLTVConfig) => async ({
         .trim() || undefined
   }))
 
+  const relatedEvents = toArray($('.related-event')).map(eventEl => ({
+    name: eventEl.find('.event-name').text(),
+    id: Number(
+      eventEl
+        .find('a')
+        .attr('href')
+        .split('/')[2]
+    )
+  }))
+
+  function findEventByName(eventName, relatedEvents): Event | undefined {
+    if (eventName == null) return undefined
+    for (var event of relatedEvents) {
+      if (event.name == eventName) {
+        return event
+      }
+    }
+    return { name: eventName }
```

Comment:
I think it's best if we return undefined instead of returning an event without an `id`.

Reply:
@gigobyte I looked at `Event` and `id` seems optional. If `eventName` is `null`, we're returning `undefined` in the beginning of the method

Reply:
`id` is optional because for compatibility reasons some old cs 1.6 events don't have pages.

I noticed some events have a "qualifies for" text that is not really an event, for example: [NoxFire League Season 2 has "LAN Finals"](https://www.hltv.org/events/4409/noxfire-league-season-2) and I don't think that should be returned as it's junk data.

Reply:
@gigobyte I mean, it's qualifying for the `LAN Finals` event, right? There is no other prize money for spots 1-4. I would rather return this, than hiding it.

This field seems to be a `string` though on hltv.org, so admins can add free-type text instead of linking it to a proper event.

Reply:
I'm up for changing the name/logic of this if you feel like there's cases that make this field logic to be something else.

Reply:
As a developer I cannot connect "LAN Finals" to anything (via id or name), making it useless for anything other than displaying a string. The real event name is "NoxFire League Season 2 Finals" so if I save this data to a DB to make some analysis later, by going with this return value we force whoever is consuming this API to check for junk event names.

Reply:
@gigobyte Right, so instead of dropping the information maybe instead of returning `Event | undefined` this should be `Event | string | undefined`?

Reply:
Yep, good thinking, I ran a script that parsed a couple hundred events and here's what I found:

"Legends" - https://www.hltv.org/events/2720/pgl-major-krakow-2017
"Pro contracts" - https://www.hltv.org/events/2870/gamerz-season-1-nordic-edition
"750$ hardware" - https://www.hltv.org/events/2960/tuddinn-spring-2017-lan-finals
"Invite to next event" - https://www.hltv.org/events/3290/tuddinn-spring-2017-lan-finals
"Withdrew" - https://www.hltv.org/events/3320/tuddinn-spring-2017-lan-finals
"100 Go4 points" - https://www.hltv.org/events/3920/tuddinn-spring-2017-lan-finals

Reply:
Yeah I just came across [2870](https://www.hltv.org/events/2870/gamerz-season-1-nordic-edition) as well. So better to change it to something else. `otherPrize` maybe?

Reply:
Sounds good. We can still have both `qualifiesFor` and `otherPrize` where qualifiesFor is a confirmed event with an id.

Reply:
@gigobyte Ok, I've done the changes. Had to make a small refactor to reduce lookups. Also added 3 more events to the tests that match this case.